File: app/src/main/java/com/example/mvvmframe/network/HeaderInterceptor.java
package com.example.mvvmframe.network;

import com.example.mvvmframe.bean.UserInfoCache;
import java.io.IOException;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * create by libo
 * create on 2018/11/13
 * description 统一添加header的拦截器
 */
public class HeaderInterceptor implements Interceptor {

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request.Builder builder = chain.request().newBuilder();
        builder.addHeader("token", UserInfoCache.TOKEN);

        return chain.proceed(builder.build());
    }
}

File: base/src/main/java/com/example/base/network/net/HeaderInterceptor.kt
package com.example.base.network.net

import com.example.base.network.bean.UserInfoCache
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

/**
 * create by libo
 * create on 2018/11/13
 * description 统一添加header的拦截器
 */
class HeaderInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val builder = chain.request().newBuilder()
        builder.addHeader("token", UserInfoCache.TOKEN)
        return chain.proceed(builder.build())
    }
}